boards: x86: disable werror for grub build

maxbachmann opened the pull request:
There is a warning because of some pointer arithmetics in Grub, that makes the build fail.
Since the build works fine without -Werror, this PR disables it.

galak commented:
> There is a warning because of some pointer arithmetics in Grub, that makes the build fail.
> Since the build works fine without -Werror, this PR disables it.

what is the error?

maxbachmann commented:
```
le-raid5_recover.o -MD -MP -MF disk/.deps-core/raid5rec_module-raid5_recover.Tpo -c -o disk/raid5rec_module-raid5_recover.o `test -f 'disk/raid5_recover.c' || echo './'`disk/raid5_recover.c
disk/mdraid1x_linux.c: In function ‘grub_mdraid_detect’:
disk/mdraid1x_linux.c:181:15: error: array subscript <unknown> is outside array bounds of ‘grub_uint16_t[0]’ {aka ‘short unsigned int[]’} [-Werror=array-bounds]
  181 |      (char *) &sb.dev_roles[grub_le_to_cpu32 (sb.dev_number)]
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
disk/mdraid1x_linux.c:98:17: note: while referencing ‘dev_roles’
   98 |   grub_uint16_t dev_roles[0]; /* Role in array, or 0xffff for a spare, or 0xfffe for faulty.  */
      |                 ^~~~~~~~~
disk/mdraid1x_linux.c:127:33: note: defined here ‘sb’
  127 |       struct grub_raid_super_1x sb;
      |                                 ^~
cc1: all warnings being treated as errors

```
This is the error message

maxbachmann committed:
boards: x86: disable werror for grub build

There is a warning because of some pointer arithmetics in Grub,
that makes the build fail. The build works after disabling -Werror.

Signed-off-by: Maximilian Bachmann <m.bachmann@acontis.com>
